packages/osd-stylelint-plugin-stylelint/src/utils/get_message.ts

```diff
@@ -15,4 +15,10 @@ export const getUntrackedMessage = (nodeInfo: { selector: string; prop: string;
 export const getTrackedMessage = (nodeInfo: { selector: string; prop: string; value: string }) =>
   `Tracked but missing approval: "${nodeInfo.selector}.${nodeInfo.prop}: ${nodeInfo.value}"`;
 
-export const getNotCompliantMessage = (message: string) => `${message}`;
+export const getNotCompliantMessage = (message: string, explanation?: string) => {
+  if (explanation) {
+    return `${message} ${explanation}`;
+  }
+
+  return message;
+};
```

Comment:
This implementation assumes `message` ends with some natural separator, i.e. with a period. Is that good enough, or should it use some more complex logic?

Reply:
Yeah, let's take this opportunity to standardize the formatting of these messages. We currently have 2 pieces:

1. A message that describes the violation
2. An explanation for why that rule exists and what to do instead.

Here's what we'd output based on the the current PR and rule definitions:

- Modifying global selector ".osdContainer" not allowed.
- Usage of property "font-family" is not allowed. OUI sets font families.
- Usage of value "#dedede" is not allowed. Use OUI variables instead of custom colors.

Suggested:

- Modifying the global selector ".osdContainer" is not allowed.
- Specifying the "font-family" property is not allowed. All font family styles should be inherited from OUI themes and components. Remove the rule.
- Using the value "#dedede" is not allowed. All colors should be inherited from the OUI theme, not hard-coded. Either remove the custom color rule or use OUI SASS variables instead.

The difference is mostly to put them all in the same verb tense and make the explanations actionable and understandable to a dev who may not even know such restrictions exist. I'm fine with the sentence formats for now - it's straightforward enough.